src/lucky/tags/text_helpers.cr

```diff
@@ -0,0 +1,233 @@
+module Lucky::TextHelpers
+  @@_cycles = Hash(String, Cycle).new
+
+  def truncate(text : String, length : Int32 = 30, omission : String = "...", separator : String | Nil = nil, escape : Bool = false, blk : Nil | Proc = nil)
+    if text
+      content = truncate_text(text, length, omission, separator)
+      content = escape ? HTML.escape(content) : content
+      content += blk.call.to_s if blk.is_a?(Proc) && text.size > length
+      content
+    end
+  end
+
+  def truncate(text : String, length : Int32 = 30, omission : String = "...", separator : String | Nil = nil, escape : Bool = true, &block : -> _)
+    truncate(text, length, omission, separator, escape, blk: block)
+  end
+
+  # This could go in a String extension
+  private def truncate_text(text : String, truncate_at : Int32, omission : String = "...", separator : String | Nil = nil)
+    return text unless text.size > truncate_at
+
+    length_with_room_for_omission = truncate_at - omission.size
+    stop = \
+      if separator
+        text.rindex(separator, length_with_room_for_omission) || length_with_room_for_omission
+      else
+        length_with_room_for_omission
+      end
+
+    "#{text[0, stop]}#{omission}"
+  end
+
+  def highlight(text : String, phrases : String | Regex | Array(String | Regex), highlighter : Proc | String = "<mark>\\1</mark>")
```

Comment:
Could we change this so `phrases` is `phrases : Array(String | Regex)` so you must always pass an array? That'll simplify the tests and code and I think will look cleaner: `phrases: ["some phrase"]`

Reply:
thoughts on doing that plus an overload?

```crystal
  def highlight(text : String, phrases : Array(String | Regex), highlighter : Proc | String = "<mark>\\1</mark>")
  # ...
  def highlight(text : String, phrase : String | Regex, highlighter : Proc | String = "<mark>\\1</mark>")
    phrases = [phrase]
    highlight(text, phrases, highlighter)
```